### PR TITLE
Return error when parsing invalid field values for line protocol

### LINKED ISSUE #2927: RC33: Panic when no value is passed to a field.
cpu,host=serverA value=2,valu1=2,value3=

panic: unsupported value type during encode fields: <nil>

goroutine 99 [running]:
github.com/influxdb/influxdb/tsdb.(*FieldCodec).EncodeFields(0xc20802bfb0, 0xc208102e10, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/tsdb/shard.go:571 +0xc56
github.com/influxdb/influxdb/tsdb.(*Shard).WritePoints(0xc208107940, 0xc20802be50, 0x1, 0x1, 0x0, 0x0)
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/tsdb/shard.go:145 +0x427
github.com/influxdb/influxdb/tsdb.(*Store).WriteToShard(0xc2080376c0, 0x2, 0xc20802be50, 0x1, 0x1, 0x0, 0x0)
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/tsdb/store.go:260 +0xf7
github.com/influxdb/influxdb/cluster.func·002(0x2, 0x1, 0xc20802be50, 0x1, 0x1)
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/cluster/points_writer.go:238 +0xfc
created by github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/cluster/points_writer.go:268 +0x30b

goroutine 1 [chan receive (nil chan), 3 minutes]:
main.(*Main).Run(0xc208017f60, 0xc20800a010, 0x2, 0x2, 0x0, 0x0)
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/cmd/influxd/main.go:81 +0x450
main.main()
	/Users/raju/Documents/raju/itsdata/go-projects/src/github.com/influxdb/influxdb/cmd/influxd/main.go:39 +0x1d0

### PR DESCRIPTION
**Pull request by jwilder**
This fixes a few issues when invalid field values are sent via the line protocol.  It fixes no values being sent as well as some cases of invalid numbers that could slip though which will not fail at parsing.

Fixes #2927

### CODE REVIEW

**Comment by corylanou** [comment]
Is there a test for an invalid number like `1.1.1`?  Otherwise, +1 looks good.

**Comment by jwilder** [comment]
I'll add one.

File: tsdb/points.go [hunk 1]
```diff
@@ -357,6 +380,62 @@ func scanTime(buf []byte, i int) (int, []byte, error) {
...
+}
+
+// scanNumber returns the end position within buf, start at i after
+// scanning over buf for an integer, or float.  It returns an
+// error if a invalid number is scanned.
+func scanNumber(buf []byte, i int) (int, []byte, error) {
+	start := i
+
+	// Is negative number?
+	if i < len(buf) && buf[i] == '-' {
```
Comment (otoolep) [inline tsdb/points.go]: Should some flag be set here?

File: tsdb/points.go [hunk 2]
```diff
@@ -740,8 +819,7 @@ func newFieldsFromBinary(buf []byte) Fields {
 		} else if (valueBuf[0] >= '0' && valueBuf[0] <= '9') || valueBuf[0] == '-' || valueBuf[0] == '.' {
 			value, err = parseNumber(valueBuf)
 			if err != nil {
-				fmt.Printf("unable to parse number value '%v': %v\n", string(valueBuf), err)
-				value = float64(0)
+				panic(fmt.Sprintf("unable to parse number value '%v': %v", string(valueBuf), err))
```
Comment (otoolep) [inline tsdb/points.go]: Are you sure you want to panic?

Comment (jwilder) [inline tsdb/points.go]: Yes, should never get here.  Would indicate we have a bug in the parser.

File: tsdb/points.go (hunk 1 above)
Comment (jwilder) [inline tsdb/points.go]: No.  Just skipping over the `-` if it starts with one.  If we see a `-` again, it'll return an error unless it's a number in scientific notation like `1.0e-3`.

Comment (otoolep) [inline tsdb/points.go]: Ah, OK, this function is just about scanning the token, not interpreting it.

Comment (jwilder) [inline tsdb/points.go]: Correct

**Comment by jwilder** [comment]
Added more tests for numbers and booleans as well as returning the line that failed in the error when parsing a batch of points.

**Comment by otoolep** [comment]
+1

**Pull request merged**